flow360/component/simulation/migration/BETDisk.py

```diff
@@ -0,0 +1,218 @@
+        for chord in updated_bet_dict["chords"]
+    ]
+
+    updated_bet_dict["sectional_polars"] = [
+        {
+            key_mapping.get(key, key): value
+            for key, value in polars.items()
+            if key not in keys_to_remove
+        }
+        for polars in updated_bet_dict["sectional_polars"]
+    ]
+
+    updated_bet_dict["alphas"] = updated_bet_dict["alphas"] * u.deg  # pylint: disable = no-member
+    updated_bet_dict["omega"] = (
+        updated_bet_dict["omega"] * u.rad / time_unit  # pylint: disable = no-member
+    )
+    updated_bet_dict["chord_ref"] = updated_bet_dict["chord_ref"] * mesh_unit
+    updated_bet_dict["sectional_radiuses"] = updated_bet_dict["sectional_radiuses"] * mesh_unit
+
+    if "blade_line_chord" in updated_bet_dict:
+        updated_bet_dict["blade_line_chord"] = updated_bet_dict["blade_line_chord"] * mesh_unit
+    return updated_bet_dict, cylinder_dict
+
+
+def _load_flow360_json(*, file_path: str) -> dict:
+    if os.path.isfile(file_path) is False:
+        raise FileNotFoundError(f"Supplied file: {file_path} cannot be found.")
+    with open(file_path, "r", encoding="utf-8") as file:
+        return json.load(file)
+
+
+@validate_call
+def read_single_v1_BETDisk(
+    file_path: str,
+    mesh_unit: LengthType.NonNegative,  # pylint: disable = no-member
+    time_unit: TimeType.Positive,  # pylint: disable = no-member
+) -> BETDisk:
+    """
+    Constructs a single :class: `BETDisk` instance from a given V1 (legacy) Flow360 input.
+
+    Parameters
+    ----------
+    file_path: str
+        Path to Flow360 JSON file that contains a **single** BETDisk setting.
+    mesh_unit: LengthType.NonNegative
+        Length unit used for LengthType BETDisk parameters.
+    time_unit: TimeType.Positive
+        Time unit used for non-dimensionalization.
+
+    Returns
+    -------
+    BETDisk
+        An instance of :class:`BETDisk` completed with given inputs.
+
+    Examples
+    --------
+    Create a BET disk with an XROTOR file.
+
+    >>> param = fl.BETDisk.from_flow360(
+    ...     file=fl.Flow360File(file_name="flow360.json")),
+    ...     mesh_unit=param.length_unit,
+    ...     time_unit=param.time_unit,
+    ... )
+    """
+
+    try:
+        bet_disk_dict, cylinder_dict = _parse_flow360_bet_disk_dict(
+            flow360_bet_disk_dict=_load_flow360_json(file_path=file_path),
+            mesh_unit=mesh_unit,
+            time_unit=time_unit,
+        )
+
+        return BETDisk(**bet_disk_dict, entities=Cylinder(**cylinder_dict))
+    except KeyError as err:
+        raise ValueError(
+            "The supplied Flow360 input for BETDisk hsa invalid format. Details: " + str(err) + "."
+        ) from err
+
+
+@validate_call
+def read_all_v1_BETDisks(
+    file_path: str,
+    mesh_unit: LengthType.NonNegative,  # pylint: disable = no-member
+    time_unit: TimeType.Positive,  # pylint: disable = no-member
```

Comment:
I don't think this is an issue, but what other value than "second" user might want to input here?